### PR TITLE
Add test coverage for TreeView

### PR DESCRIPTION
**Pull request by Zheng-Li01**
related https://github.com/dotnet/winforms/issues/10453

## Proposed changes
- Add unit tests for TreeView to test its properties: Name, IsReadOnly, ContextMenuStrip, RowBounds, IsVisible, IsExpanded, EnsureVisible
- Add unit tests for TreeView to test its methods: GetChildCount, Add, Insert, CanConvertTo, ConvertTo, HitTest, FromHandle, ExpandAll, Clone, Collapse, GetNodeCount, Toggle

## Customer Impact
- None

## Regression? 
- No

## Risk
- Minimal

## Test methodology 
- Unit tests

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/11293)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs [hunk 1]
```diff
@@ -223,4 +223,185 @@ public void TreeNodeCollection_Find_NullOrEmptyKey_ThrowsArgumentNullException(s
         Assert.Throws<ArgumentNullException>("key", () => collection.Find(key, searchAllChildren: true));
         Assert.Throws<ArgumentNullException>("key", () => collection.Find(key, searchAllChildren: false));
     }
+
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs:226]: Those tests could merged together, due to similar setup and parameters:
	- `TreeNodeCollection_Add_KeyTextImageIndex_Success`, `TreeNodeCollection_Add_KeyTextImageIndexSelectedImageIndex_Success`
	- `TreeNodeCollection_Add_KeyTextImageIndex_Success`, `TreeNodeCollection_Add_KeyTextImageIndexSelectedImageIndex_Success`
	- `TreeNodeCollection_Insert_String_Success`, `TreeNodeCollection_Insert_StringKey_Success`, `TreeNodeCollection_Insert_StringKeyImageKey_Success`, `TreeNodeCollection_Insert_StringKeyImageKeySelectedImageKey_Success`
    - `TreeNodeCollection_Insert_StringKeyImageIndex_Success`, `TreeNodeCollection_Insert_StringKeyImageIndexSelectedImageIndex_Success`

Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeCollectionTests.cs:226]: Those tests could merged together, due to similar setup and parameters:
	- `TreeNodeCollection_Add_KeyTextImageIndex_Success`, `TreeNodeCollection_Add_KeyTextImageIndexSelectedImageIndex_Success`
	- `TreeNodeCollection_Add_KeyTextImageIndex_Success`, `TreeNodeCollection_Add_KeyTextImageIndexSelectedImageIndex_Success`
	- `TreeNodeCollection_Insert_String_Success`, `TreeNodeCollection_Insert_StringKey_Success`, `TreeNodeCollection_Insert_StringKeyImageKey_Success`, `TreeNodeCollection_Insert_StringKeyImageKeySelectedImageKey_Success`
    - `TreeNodeCollection_Insert_StringKeyImageIndex_Success`, `TreeNodeCollection_Insert_StringKeyImageIndexSelectedImageIndex_Success`

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeTests.cs [hunk 2]
```diff
@@ -4632,6 +4632,233 @@ public void TreeNode_InvokeAdd_WithoutTree_DoesNotAddNodeToTrackList()
...
+        treeNode.Nodes.Add(new TreeNode(childNode2));
+        TreeNode clonedNode = (TreeNode)treeNode.Clone();
+
+        clonedNode.Text.Should().Be(treeNode.Text);
+        clonedNode.Checked.Should().Be(treeNode.Checked);
+        clonedNode.Nodes.Count.Should().Be(treeNode.Nodes.Count);
+        clonedNode.Nodes[0].Text.Should().Be(treeNode.Nodes[0].Text);
+        clonedNode.Nodes[1].Text.Should().Be(treeNode.Nodes[1].Text);
+    }
+
```
Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeTests.cs:4733]: `TreeNode_Collapse_ReturnsExpected` and `TreeNode_ExpandAll_Invoke_Success` could be combined into a single test method.

Comment (ricardobossan) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TreeNodeTests.cs:4733]: `TreeNode_Collapse_ReturnsExpected` and `TreeNode_ExpandAll_Invoke_Success` could be combined into a single test method.

**Review by lonitra — APPROVED** [review]
👍

**Pull request merged**